src/DisplayPreference.php

```diff
@@ -329,146 +337,78 @@ public function showFormPerso($target, $itemtype)
-                             " class='btn btn-icon btn-sm btn-ghost-secondary'><i class='ti ti-arrow-up'></i></button>";
-                            Html::closeForm();
-                            echo "</td>\n";
-                        } else {
-                            echo "<td>&nbsp;</td>";
-                        }
-
-                        if ($i != ($numrows - 1)) {
-                            echo "<td>";
-                            echo "<form method='post' action='$target'>";
-                            echo "<input type='hidden' name='id' value='" . $data["id"] . "'>";
-                            echo "<input type='hidden' name='users_id' value='$IDuser'>";
-                            echo "<input type='hidden' name='itemtype' value='$itemtype'>";
-                            echo "<button type='submit' name='down'" .
-                            " title=\"" . __s('Bring down') . "\"" .
-                            " class='btn btn-icon btn-sm btn-ghost-secondary'><i class='ti ti-arrow-down'></i></button>";
-                            Html::closeForm();
-                            echo "</td>\n";
-                        } else {
-                            echo "<td>&nbsp;</td>";
-                        }
-
-                        if (!isset($searchopt[$data["num"]]["noremove"]) || $searchopt[$data["num"]]["noremove"] !== true) {
-                            echo "<td>";
-                            echo "<form method='post' action='$target'>";
-                            echo "<input type='hidden' name='id' value='" . $data["id"] . "'>";
-                            echo "<input type='hidden' name='users_id' value='$IDuser'>";
-                            echo "<input type='hidden' name='itemtype' value='$itemtype'>";
-                            echo "<button type='submit' name='purge'" .
-                             " title=\"" . _sx('button', 'Delete permanently') . "\"" .
-                             " class='btn btn-icon btn-sm btn-ghost-secondary'><i class='ti ti-x'></i></button>";
-                            Html::closeForm();
-                            echo "</td>\n";
-                        } else {
-                            echo "<td>&nbsp;</td>\n";
-                        }
-                        echo "</tr>";
-                        $i++;
-                    }
-                }
-            }
-            echo "</table>";
         }
-        echo "</div>";
+        $entries = [];
+        foreach ($fixed_columns as $key => $val) {
+            $entries[] = [
+                'id'   => $val,
+                'name' => $searchopt[$val]['name'],
+                'group' => $this->nameOfGroupForItemInSearchopt($searchopt, $val),
+                'fixed' => true
+            ];
+        }
+        foreach ($already_added as $key => $val) {
+            $entries[] = [
+                'id'   => $val,
+                'name' => $searchopt[$val]['name'],
```

Comment:
Probably due to wrong data in database (removed plugin for example), but I guess you need to manage the case

```
[2023-08-23 14:11:27] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key 47812 in /var/www/html/glpi/10.1.git/src/DisplayPreference.php at line 386
  Backtrace :
 ```
and
```
[2023-08-23 14:11:27] glpiphplog.WARNING:   *** PHP Warning (2): Trying to access array offset on value of type null in /var/www/html/glpi/10.1.git/src/DisplayPreference.php at line 386
```